plugins/services/src/js/components/modals/CreateServiceModalForm.js

```diff
@@ -72,6 +72,34 @@ const CONSTANTLY_UNMUTED_ERRORS = [
   /^volumes\.[0-9]+\./
 ];
 
+function cleanConfig(config) {
+  const { labels = {}, env = {}, environment = {}, ...serviceConfig } = config;
+
+  let newServiceConfig = CreateServiceModalFormUtil.stripEmptyProperties(
```

Comment:
Would it be feasible to add a param for exceptions to the `CreateServiceModalFormUtil. stripEmptyProperties` method? I would expect something like this could work an be a bit easier on this side of the table

```js
stripEmptyProperties(object, blacklist = []) {
    if (typeof object !== "object" || object === null) {
      return object;
    }

    // Pick base object according to type
    let baseObject = {};
    if (Array.isArray(object)) {
      baseObject = [];
    }

    return Object.keys(object).reduce(function(memo, key) {
      if (
        (!ValidatorUtil.isEmpty(object[key]) && !Number.isNaN(object[key])) ||
        Array.isArray(object[key])
      ) {
        // Apply the strip function recursively and keep only non-empty values
        const value = CreateServiceModalFormUtil.stripEmptyProperties(
          object[key]
        );
        if (
          !blacklist.includes(key) ||
          (!ValidatorUtil.isEmpty(value) && !Number.isNaN(value)) ||
          Array.isArray(value)
        ) {
          memo[key] = value;
        }
      }

      return memo;
    }, baseObject);
  }
};
```

Reply:
That is actually a neat idea. We already kind of established a bad habit and we use a bad pattern. I'll look into this.